x-pack/legacy/plugins/siem/public/pages/network/index.tsx

```diff
@@ -7,31 +7,74 @@
+import { GlobalTime } from '../../containers/global_time';
+import { IpDetailsTableType } from '../../store/network/model';
 import { Network } from './network';
+import { AnomaliesQueryTabBody, TlsQueryTabBody, UsersQueryTabBody } from './navigation';
 
-const networkPath = `/:pageName(network)`;
+const networkPagePath = `/:pageName(network)`;
+const ipDetailsPagePath = networkPagePath + '/ip/:detailName';
 
 type Props = Partial<RouteComponentProps<{}>> & { url: string };
 
+const getIPDetailsTabPath = (pagePath: string) =>
+  `${pagePath}/ip/:detailName/:tabName(` +
+  `${IpDetailsTableType.tls}|` +
+  `${IpDetailsTableType.users}|` +
+  `${IpDetailsTableType.anomalies})`;
+
 export const NetworkContainer = React.memo<Props>(() => (
-  <Switch>
-    <Route strict exact path={networkPath} render={() => <Network />} />
-    <Route
-      path={`${networkPath}/ip/:detailName`}
-      render={({
-        match: {
-          params: { detailName },
-        },
-      }) => <IPDetails detailName={detailName} />}
-    />
-    <Route
-      path="/network/"
-      render={({ location: { search = '' } }) => (
-        <Redirect from="/network/" to={`/network${search}`} />
-      )}
-    />
-  </Switch>
+  <GlobalTime>
+    {({ to, from, setQuery, deleteQuery, isInitializing }) => (
+      <Switch>
+        <Route strict exact path={networkPagePath} render={() => <Network />} />
+        <Route
+          strict
+          exact
+          path={getIPDetailsTabPath(networkPagePath)}
+          render={props => {
+            const commonProps = {
+              deleteQuery,
+              from,
+              to,
+              setQuery,
+              isInitializing,
+              detailName: props.match.params.detailName,
+            };
+
+            return (
+              <>
+                <IPDetails {...commonProps} />
+                <Route
+                  path={`${ipDetailsPagePath}/:tabName(${IpDetailsTableType.tls})`}
+                  render={() => <IPDetailsBody {...commonProps} children={TlsQueryTabBody} />}
+                />
+                <Route
+                  path={`${ipDetailsPagePath}/:tabName(${IpDetailsTableType.users})`}
+                  render={() => <IPDetailsBody {...commonProps} children={UsersQueryTabBody} />}
+                />
+                <Route
+                  path={`${ipDetailsPagePath}/:tabName(${IpDetailsTableType.anomalies})`}
+                  render={() => <IPDetailsBody {...commonProps} children={AnomaliesQueryTabBody} />}
+                />
+              </>
+            );
+          }}
+        />
+        <Route
+          path={`${networkPagePath}/ip/:detailName`}
+          render={({ location: { search = '', pathname } }) => (
+            <Redirect to={`${pathname}/${IpDetailsTableType.tls}${search}`} />
+          )}
+        />
+        <Route
+          path="/network/"
+          render={({ location: { search = '' } }) => <Redirect to={`/network${search}`} />}
```

Comment:
We could replace the hard coded `network` with `SiemPageName.network;` here

Reply:
Thank you 👍